src/components/SwitzerlandEstimatedCasesDivisionModal.tsx

```diff
@@ -87,26 +88,23 @@ export const SwitzerlandEstimatedCasesDivisionModal = ({
   show,
   handleClose,
 }: Props) => {
-  const [caseCounts, setCaseCounts] = useState<CaseCountEntry[] | undefined>();
+  //const [caseCounts, setCaseCounts] = useState<CaseCountEntry[] | undefined>();
   const [showSwissRegions, setShowSwissRegions] = useState(true);
-
   const { dateFrom, dateTo, country } = variantSampleSet.sampleSelector;
-  useEffect(() => {
-    let isSubscribed = true;
-    const controller = new AbortController();
-    const signal = controller.signal;
 
-    getCaseCounts({ dateFrom, dateTo, country }, true, signal).then(caseCounts => {
-      if (isSubscribed) {
-        setCaseCounts(caseCounts);
-      }
-    });
+  const fetchCaseCounts = async () => {
+    const data = await getCaseCounts({ dateFrom, dateTo, country }, true, new AbortController().signal);
+    return data;
+  };
 
-    return () => {
-      isSubscribed = false;
```

Comment:
@chaoran-chen @TKGZ What is the purpose of `isSubscribed` ?

Reply:
This prevents duplicate fetches, for example if a new fetch is called before the previous one finishes.

Reply:
@ningxie1991, we have some queries that can take quite some time. Do you know what will now happen if the user navigates to another page before the data for the first one was fully loaded. I.e. if the first query is finished after the second query?

Reply:
I updated the code so that react query behaves the same. It will cancel request if we close or click out of the modal. And `refetch` data in `useEffect` if `isFetching` = false 

![image](https://user-images.githubusercontent.com/3387698/130066887-9d50fac9-4217-4a95-8fd3-face5e4903a9.png)